netket/nn/linear.py

```diff
@@ -314,6 +318,95 @@ def __call__(self, inputs: Array) -> Array:
         return y
 
 
+class DenseEquivariant(Module):
+    """An equivariant layer that takes a feature map with embeddings corresponding to various
+    poses of a symmetry group, shape = [n_symm*in_features] and outputs a feature map of identical
+    shape [n_symm*out_features]. This transformation respects the group structure such that
+    g L({f}) = L(g{f}) where g is an element of the symmetry group and {f} are the feature embeddings.
```

Comment:
Maybe you could also give an explicit example of what happens to shapes. 
Given an input shape (N_batches, N_spins, blablabla) you get ...